subsys/bluetooth/host/id.c

```diff
@@ -569,8 +569,6 @@ static void rpa_timeout(struct k_work *work)
 {
 	bool adv_enabled = false;
 
-	BT_DBG("");
-
```

Comment:
Empty debug statements can still be useful. Why is this causing a compile error?

Reply:
It generates the following warning
`warning: zero-length gnu_printf format string [-Wformat-zero-length]`

Reply:
We currently have 145 cases of `BT_DBG("");` in the Bluetooth subsystem. This must not cause any compile errors or warnings. If it does for this specific file, then something is odd, and should be investigated

Reply:
since there's no kconfig, could it be `CONFIG_LOG_FUNC_NAME_PREFIX_DBG` that is not set? could you check if setting it fixes the issue @ahmedmoheb-nordic ?

Reply:
> since there's no kconfig, could it be `CONFIG_LOG_FUNC_NAME_PREFIX_DBG` that is not set? could you check if setting it fixes the issue @ahmedmoheb-nordic ?

Unfortunately, it has no effect

Reply:
@ahmedmoheb-nordic Which SDK/toolchain and version are you using to build? I mean, we are building this in CI every day without issues, right?

Reply:
@Thalley `zephyr-sdk-0.15.0` `GNU 9.4.0`

Reply:
Interesting. If you build the `main` branch, does it appear as well? 

I'm just curious as how you managed to get the warning :)

Reply:
To be honest, I don't know even how to test it.
Is there a script to run or something like that? Thanks to help with trying that.

Reply:
> To be honest, I don't know even how to test it. Is there a script to run or something like that? Thanks to help with trying that.

How do you mean? Did you not compile it locally to get the warning/error?

Reply:
> > To be honest, I don't know even how to test it. Is there a script to run or something like that? Thanks to help with trying that.
> 
> How do you mean? Did you not compile it locally to get the warning/error?

Oh, sorry I thought that you were asking about building other projects that are using id.c, sorry for the missunderstanding.

From my side, yes, I'm able to build my unit test projects locally to get such error.
Just by commenting that line, it works just fine.

Reply:
@Thalley I want to ask if all build are using the same set of compilation options/flags.
Is there a possibility that for unit testing there a different set of compilation options?

Reply:
Ah, you are building it as part of doing unittests? That explains why I haven't seen this issue :) 

So this warning seems to be related to how unittests are built, and would perhaps also explain why we are not seeing any issues with the other occurences of `BT_DBG("")`, as most of the files containing those, do not have unittests. 

In any case, it sounds to me that this is an issue that needs to be solved in how the unittests are built/handles `BT_DBG("")`, and not something that needs to be fixed in the host stack.

Reply:
@Thalley I agree with your point to investigate/resolve the issue with unit tests, but that is going to take some time before someone handles it.

For now, that's blocking the UTs, what about suppressing that compilation option for my UT project by adding the following to `CMakelist.txt`

`# Supress the format-zero-length error if GNUC is used
if (CMAKE_C_COMPILER_ID MATCHES "GNU") 
add_compile_options(-Wno-error=format-zero-length -Wno-format-zero-length)
endif() `

what do you think?

Reply:
I think that's fine for getting the test in, but you should get to the bottom of it when there is another UT that needs it, and remove that workaround then.

Reply:
@Thalley I dropped that change, please re-check